### DISCUSSION TITLE
server side global state that's request-private?

### DISCUSSION THREAD

**Discussion by voideanvalue** [vercel/next.js#35709] [Help]
where request = execution context for rendering of a page (whether at build time for SSG, or at runtime for SSR)

Basically, I want to ensure that a react-external server-side global state store instance<sup>1</sup> is _never_ shared across requests.

Looking through examples, I've come across variations of this pattern: https://github.com/vercel/next.js/blob/canary/examples/with-redux/src/app/store.ts#L24
https://github.com/vercel/next.js/blob/canary/examples/with-redux/src/pages/_app.tsx#L6,L10

but that seems flawed as the `app/store.ts` module factory executes once for the lifetime of the server process... so all requests will end up sharing the exported same store instance. Previous requests potentially leaking request/user sensitive data to subsequent ones is scary.

I tried doing something like `initStore` inside each page's `getStaticProps` to get a fresh store instance for each request/page... but then I got blocked when it came to accessing the same instance in `pages/_app`.

I tried threading it via `props` returned from each page's `getStaticProps` to `pageProps` passed to the react component in `pages/_app`... but that requires the value being serializable.

I also tried looking for request/page lifecycle methods that I could leverage and came up empty.

Is there some way to achieve this with next.js?

<sub>1: that's a lot of qualifiers :P ... I couldn't figure out how else to phrase this. suggestions welcome!</sub>

**Comment by nimeshvaghasiya** [comment]
I need something that make api call once per user login and share data across all pages, HOC(higher order component). 
**should only be on server side, so that if user has no permission to view the page, I can redirect back to login**
